### PR TITLE
Add support for DLS in Dropbox

### PR DESCRIPTION
**Pull request by akanshi-elastic**
#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/dropbox.py [hunk 1]
```diff
@@ -753,6 +966,95 @@ async def advanced_sync(self, rule):
                 else:
                     yield self._adapt_dropbox_doc_to_es_doc(response=data), data
 
+    def get_id(self, permission, identity):
```
Comment (vidok) [inline connectors/sources/dropbox.py:969]: should the method name be `group_id`? to match the property it pulls?

File: connectors/sources/dropbox.py [hunk 2]
```diff
@@ -763,18 +1065,57 @@ async def get_docs(self, filtering=None):
...
-                    return document, partial(self.get_content, attachment=attachment)
+                    return document, partial(
+                        self.get_content, attachment=attachment, folder_id=folder_id
+                    )
             else:
                 return document, None
 
-        if filtering and filtering.has_advanced_rules():
+        if self._dls_enabled():
+            account_id, member_id = await self.get_account_details()
```
Comment (vidok) [inline connectors/sources/dropbox.py:1085]: It's better to decouple this block of code into separate method.

**Comment by akanshi-elastic** [comment]
Will update the PR soon with suggested change:https://crestdatasystems.slack.com/archives/C02AV5C8ZDE/p1698684375821709?thread_ts=1698657367.352959&cid=C02AV5C8ZDE

File: connectors/sources/dropbox.py [hunk 3]
```diff
@@ -35,7 +40,10 @@
 DEFAULT_RETRY_AFTER = 300  # seconds
 RETRY_INTERVAL = 2
 MAX_CONCURRENT_DOWNLOADS = 100
+REQUEST_BATCH_SIZE = 50
```
[the remark this answers is not in the document]
Comment (akanshi-elastic) [reply connectors/sources/dropbox.py:43]: max batch size we can pass is 100 as per [documentation](https://www.dropbox.com/developers/documentation/http/documentation#sharing-list_file_members-batch:~:text=files%20List%20of%20(String(min_length%3D1%2C%20pattern%3D%22((/%7Cid%3A).*%7Cnspath%3A%5B0%2D9%5D%2B%3A.*)%7Cns%3A%5B0%2D9%5D%2B(/.*)%3F%22)%2C%20max_items%3D100)Files%20for%20which%20to%20return%20members.) but as we are storing the complete document with the file id in dictionary on line 1137 so it's throwing  `Queue Full` error while running the connector. To fix this issue we tried to pass less number document in batch and it's working fine with `REQUEST_BATCH_SIZE` 50

**Review by vidok — APPROVED** [review]

**Pull request merged**